src/biokbase/narrative/jobs/appmanager.py

```diff
@@ -466,41 +415,281 @@ def run_app(
+
+        Parameters:
+        -----------
+        app_info: this is a list of app information dictionaries. It's broken down such that a single app
+            can have multiple sets of parameters, which could create multiple runs of that app.
+            Each dictionary is expected to have the following keys:
+            app_id: the id of the app to run
+            tag: the app tag to run, one of release, beta, or dev
+            version: (optional) the specified version to run, if not provided, this will be the most recent
+                for that particular tag
+            params: a list of at least dictionary. Each dict contains the set of parameters to run the
+                app once.
+        cell_id: if provided, this should be a unique id for the Narrative cell that's running the app.
+        run_id: if provided, this should be a unique id representing a Narrative cell's knowledge of
+            that job.
+        dry_run: if True, this won't start the job, but return the structure that would be sent to the
+            KBase execution engine.
+
+        Example:
+        --------
+        run_app_batch([{
+            "app_id": "MegaHit/run_megahit",
+            "tag": "release",
+            "version": "1.0.0",
+            "params": [{
+                    "read_library_name" : "My_PE_Library",
+                    "output_contigset_name" : "My_Contig_Assembly"
+            }]
+        }])
+        """
+
+        if not isinstance(app_info, list) or len(app_info) == 0:
+            raise ValueError(
+                "app_info must be a list with at least one set of app information"
+            )
+        batch_run_inputs = list()
+        ws_id = strict_system_variable("workspace_id")
+        batch_params = {"wsid": ws_id}  # for EE2.run_job_batch
+        log_app_info = list()
+        for info in app_info:
+            self._validate_bulk_app_info(info)
+            app_id = info["app_id"]
+            tag = info.get("tag", "release")
+            version = info.get("version")
+            spec = self._get_validated_app_spec(app_id, tag, True, version)
+            for param_set in info["params"]:
+                # will raise a ValueError if anything is wrong or missing
+                # otherwise, will build a set of inputs for EE2.run_job
+                batch_run_inputs.append(
+                    self._build_run_job_params(
+                        spec,
+                        tag,
+                        param_set,
+                        version=version,
+                        cell_id=cell_id,
+                        run_id=run_id,
+                        ws_id=ws_id,
+                    )
+                )
+            log_app_info.append(
+                {
+                    "app_id": app_id,
+                    "tag": tag,
+                    "version": version,
+                    "num_jobs": len(batch_run_inputs),
```

Comment:
is this designed to be an accumulator, listing the number of jobs so far?

Reply:
Sort of. That'll wind up being a list with one element for each app, showing how many jobs of that app are being run. That list then gets passed to the "run_app_bulk" log statement.

Reply:
I can't quite tell where it is indented to and thus what the value is going to be -- that's the only reason I ask.